cli/bin/exec.js

```diff
@@ -17,10 +17,28 @@ function getGrainc() {
 
 const grainc = getGrainc();
 
-function exec(commandOrFile = "", options = {}, execOpts = { stdio: "pipe" }) {
-  let cflags = options.cflags ? options.cflags : "";
-  let stdlib = options.stdlib ? `--stdlib=${options.stdlib}` : "";
-  return execSync(`${grainc} ${stdlib} ${cflags} ${commandOrFile}`, execOpts);
+function exec(commandOrFile = "", program, execOpts = { stdio: "pipe" }) {
+  const optionValues = program.opts();
+
+  const flags = [];
+  program.options.forEach((option) => {
+    if (!option.grainc) return;
+
+    const value = optionValues[option.attributeName()];
+
+    if (value instanceof Array && value.length > 0) {
+      flags.push(`${option.long || option.short} ${value.join(",")}`);
+    } else if (typeof value === "string" || typeof value === "number") {
+      flags.push(`${option.long || option.short} ${value}`);
+    } else if (
+      (option.negate && value === false) ||
+      (!option.negate && value === true)
+    ) {
+      flags.push(option.long || option.short);
+    }
```

Comment:
Should we emit some sort of warning if there's a flag whose type we don't understand?